internal/pkg/utils/utils.go

```diff
@@ -0,0 +1,21 @@
+package utils
```

Comment:
So `utils` isn't a preferred package name. Scroll down to the section on "Bad Package Names" in [this article](https://blog.golang.org/package-names).

```
Packages named util, common, or misc provide clients with no sense of what the package contains. This makes it harder for clients to use the package and makes it harder for maintainers to keep the package focused.
```

I would be fine if you just put this function in an `encoding.go` file right in the `internal/pkg` folder. No need for a sub-package. Also add the appropriate copyright block to the top of this new file.

Also, for future reference when you have a package and then a Go file in that package with the same name -- `utils/utils.go` -- this is known as "stutter" and should be avoided.